### PR TITLE
Database creation

### PR DESCRIPTION
**Pull request by Hussein-shahatet97**

### CODE REVIEW

File: server/database/config/db_build.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const fs = require('fs');
+const path = require('path');
+
+const dbConnection = require('./db_Connection');
```
Comment (karmelyoei) [inline server/database/config/db_build.js]: Please check the db connection name of the file it supposes to be like this "db_connection"

**Review by karmelyoei — COMMENTED** [review]
Please check the file name of the db connection

**Review by karmelyoei — COMMENTED** [review]
Where is the fake dummy data?

File: server/database/config/db_build.sql [hunk 2]
```diff
@@ -0,0 +1,67 @@
+BEGIN;
+  DROP TABLE IF EXISTS employees,questions,employees_questions,forms,employee_form,form_questions
+  CASCADE;
+
```
Comment (yaqootturman) [inline server/database/config/db_build.sql:4]: Let`s rename ids to contain his table name, id for question table make it question_id and so on

File: server/database/config/db_build.sql [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+CREATE TABLE employees
+(
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(50) NOT NULL,
+  email VARCHAR  ,
+  image TEXT,
```
Comment (yaqootturman) [inline server/database/config/db_build.sql]: bytea is the data type for an image, why using TEXT? if you mean to use a URL as a source we can start as a bytea and store it as an image from the beginning

**Review by yaqootturman — CHANGES_REQUESTED** [review]

File: server/database/config/db_build.sql [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  role VARCHAR
+);
+
+CREATE TABLE questions
+(
+  id SERIAL PRIMARY KEY,
+  employee_id INT,
+  FOREIGN KEY (employee_id)  REFERENCES employees(id),
+  type VARCHAR NOT NULL,
+  result_rate VARCHAR ,
```
Comment (yaqootturman) [inline server/database/config/db_build.sql:24]: I see the result-rate should be INT, but if you have a reason to choose VARCHAR let`s discuss, please!

File: server/database/config/db_build.sql [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  context TEXT
```
Comment (yaqootturman) [inline server/database/config/db_build.sql:23]: another thing, we see to add an answer column in the question table, true?

**Review by yaqootturman — CHANGES_REQUESTED** [review]

File: server/database/config/db_build.js (hunk 1 above)
Comment (Hussein-shahatet97) [reply server/database/config/db_build.js]: done !

File: server/database/config/db_build.sql (hunk 3 above)
Comment (Hussein-shahatet97) [reply server/database/config/db_build.sql]: this type has been modified.

File: server/database/config/db_build.sql (hunk 4 above)
Comment (Hussein-shahatet97) [reply server/database/config/db_build.sql:24]: this data type has been changed .

File: server/database/config/db_build.sql (hunk 5 above)
Comment (Hussein-shahatet97) [reply server/database/config/db_build.sql:23]: this column has been added.

File: server/database/config/db_build.sql (hunk 2 above)
Comment (Hussein-shahatet97) [reply server/database/config/db_build.sql:4]: I think the correct way to use identifiers as primary keys in their tables is to type as they are, but when we need them as foreign keys, we should write them in relation to their table names.

**Review by yaqootturman — DISMISSED** [review]

**Review by rubasider — DISMISSED** [review]

**Comment by FarahZaqout** [comment]
You are linking the employee id with the form id twice.
First time via the intermediate table. Second time directly. I don't understand the need to link it a second time when you have the intermediate table.

@Hussein-shahatet97

**Review by yaqootturman — APPROVED** [review]

**Review by rubasider — APPROVED** [review]

**Pull request merged**